pkg/trustedresources/verify.go

```diff
@@ -239,3 +353,18 @@ func parseRef(k8sRef string) (string, string, error) {
 	}
 	return s[0], s[1], nil
 }
+
+// hashAlgorithm returns a crypto.Hash code using an algorithm name as input parameter
+func hashAlgorithm(algorithmName string) (crypto.Hash, error) {
+	if algorithmName == "" {
+		return crypto.SHA256, nil
+	}
+	normalizedAlgo := strings.ToLower(strings.TrimSpace(algorithmName))
```

Comment:
im curious about what motivation led to normalizing the algorithm name like this? i would have thought we'd just consider names that didn't exactly match invalid, im wondering what the benefit is here (and it seems like we allow extra spaces as well?)

unless we have a clear reason id prefer we start out less permissive, i.e. require the exact expected strings since it's easier to relax restrictions later than go the other way

Reply:
it's used for allow different cases of algorithms. e.g. sha256, Sha256, SHA256 should all be valid. It's mainly for better user experience